src/main/org/codehaus/groovy/transform/stc/StaticTypeCheckingSupport.java

```diff
@@ -1157,6 +1142,45 @@ private static ClassNode makeRawType(final ClassNode receiver) {
         return result;
     }
 
+    private static void removeMethodInSuperInterface(List<MethodNode> toBeRemoved, MethodNode one, MethodNode two) {
+        ClassNode oneDC=one.getDeclaringClass();
+        ClassNode twoDC=two.getDeclaringClass();
+        if(oneDC.implementsInterface(twoDC)){
+            toBeRemoved.add(two);
+        }else{
+            toBeRemoved.add(one);
+        }
+    }
+
+    private static boolean areEquivalentInterfaceMethods(MethodNode one, MethodNode two, Parameter[] onePars, Parameter[] twoPars) {
```

Comment:
I think it would simplify things if this just took 2 MethodNode's, the parameters can be gotten from the nodes so no need to pass them in I think?